### DISCUSSION TITLE
Edge: Is the @section and @layout tags still available and working?

### DISCUSSION THREAD

**Discussion by ekimkael** [adonisjs/core#4358] [Help]
Hi guys,
I tried Adonisjs version 6 yesterday but as I was reading the doc of Edge I didn't found `@section` and `@layout` tags.
I tried to use them and it didn't work. Can you please redirect me to the new way of handling layouts please?

**Comment by RomainLanz** [comment]
Hey @ekimkael! 👋🏻 

The `@layout` feature [has been removed in Edge 6](https://edgejs.dev/docs/changelog/upgrading-to-v6#breaking-changes).
As stated in the release note, you can install a compatibility layer if you have to continue to use those.

We recommend using the component API from now on.

```html
<!-- resources/views/components/layouts/main.edge -->

<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    
    <!-- ... -->
  </head>
  <body>
    {{{ await $slots.main() }}}
  </body>
</html>
```

```html
<!-- resources/views/pages/home.edge -->

@layouts.main()
  <h1>Hello World</h1>
@end
```

**Reply by ekimkael** [reply]
Is the example below the new way of doing things?
Because it's still doesn't work when for this case:
the layout is here: `<!-- resources/views/layouts/app.edge -->`
the page here: `<!-- resources/views/places/index.edge -->`

Should these files be in the exact directories as yours?

Also for the compatibility where should I set/copy this code please?
The doc doesn't say where

**Reply by thetutlage** [reply]
Doc doesn't have to teach you where to store files. You store them as you like and reference them from the file path.

Also, if you need help, you will have to share the exact code and the exact errors vs just snippets of code.

---

Also docs does mention the removal of layouts in favor of components and seems like you missed that with the option to use the compatibility plugin to get the layouts back.

<img width="725" alt="Screenshot 2024-01-27 at 10 09 35 AM" src="https://github.com/adonisjs/core/assets/1706381/97fa3244-cc84-415e-be3a-62f5031d4205">

**Reply by ekimkael** [reply]
I didn't meant to be rude.
I was just asking, I've read the doc that's why I asked if there was a folder where I can add the compatibility plugin.

**Comment by Esames786** [comment]
@component('main/layout/master')
@set('title', 'Dashboard')
@layouts.main()
  <h1>test</h1>
@end
@endcomponent
